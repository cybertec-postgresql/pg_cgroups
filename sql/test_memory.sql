-- check the default settings (should both be -1)
SHOW pg_cgroups.memory_limit;
SHOW pg_cgroups.swap_limit;
SHOW pg_cgroups.oom_killer;

-- change swap_limit (will set the parameter, leave kernel value unlimited)
ALTER SYSTEM SET pg_cgroups.swap_limit = 512;
SELECT pg_reload_conf();
SELECT pg_sleep_for('0.3');
SHOW pg_cgroups.swap_limit;

-- change memory limit (should work)
ALTER SYSTEM SET pg_cgroups.memory_limit = 1024;
SELECT pg_reload_conf();
SELECT pg_sleep_for('0.3');
SHOW pg_cgroups.memory_limit;
SHOW pg_cgroups.swap_limit;

-- change swap_limit (should work)
ALTER SYSTEM SET pg_cgroups.swap_limit = 0;
SELECT pg_reload_conf();
SELECT pg_sleep_for('0.3');
SHOW pg_cgroups.swap_limit;

-- lower memory limit (should work)
ALTER SYSTEM SET pg_cgroups.memory_limit = 256;
SELECT pg_reload_conf();
SELECT pg_sleep_for('0.3');
SHOW pg_cgroups.memory_limit;
SHOW pg_cgroups.swap_limit;

-- raise memory limit (should work)
ALTER SYSTEM SET pg_cgroups.memory_limit = 512;
SELECT pg_reload_conf();
SELECT pg_sleep_for('0.3');
SHOW pg_cgroups.memory_limit;
SHOW pg_cgroups.swap_limit;

-- set swap limit to -1 (should work)
ALTER SYSTEM SET pg_cgroups.swap_limit = -1;
SELECT pg_reload_conf();
SELECT pg_sleep_for('0.3');
SHOW pg_cgroups.swap_limit;

-- set swap limit to 0 (should work)
ALTER SYSTEM SET pg_cgroups.swap_limit = 0;
SELECT pg_reload_conf();
SELECT pg_sleep_for('0.3');
SHOW pg_cgroups.swap_limit;

-- set memory limit to 0 (should fail)
ALTER SYSTEM SET pg_cgroups.memory_limit = 0;

-- disable OOM killer (should work)
ALTER SYSTEM SET pg_cgroups.oom_killer = off;
SELECT pg_reload_conf();
SELECT pg_sleep_for('0.3');
SHOW pg_cgroups.oom_killer;

-- reset all settings
ALTER SYSTEM RESET pg_cgroups.memory_limit;
ALTER SYSTEM RESET pg_cgroups.swap_limit;
ALTER SYSTEM RESET pg_cgroups.oom_killer;
SELECT pg_reload_conf();
SELECT pg_sleep_for('0.3');
SHOW pg_cgroups.memory_limit;
SHOW pg_cgroups.swap_limit;
SHOW pg_cgroups.oom_killer;
