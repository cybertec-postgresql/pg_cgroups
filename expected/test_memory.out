-- check the default settings (should both be -1)
SHOW pg_cgroups.memory_limit;
 pg_cgroups.memory_limit 
-------------------------
 -1
(1 row)

SHOW pg_cgroups.swap_limit;
 pg_cgroups.swap_limit 
-----------------------
 -1
(1 row)

-- change swap_limit (expect no effect, because there is no memory limit)
ALTER SYSTEM SET pg_cgroups.swap_limit = 512;
SELECT pg_reload_conf();
 pg_reload_conf 
----------------
 t
(1 row)

SELECT pg_sleep_for('0.3');
 pg_sleep_for 
--------------
 
(1 row)

SHOW pg_cgroups.swap_limit;
 pg_cgroups.swap_limit 
-----------------------
 -1
(1 row)

-- change only memory limit (should work)
ALTER SYSTEM SET pg_cgroups.memory_limit = 1024;
SELECT pg_reload_conf();
 pg_reload_conf 
----------------
 t
(1 row)

SELECT pg_sleep_for('0.3');
 pg_sleep_for 
--------------
 
(1 row)

SHOW pg_cgroups.memory_limit;
 pg_cgroups.memory_limit 
-------------------------
 1024
(1 row)

SHOW pg_cgroups.swap_limit;
 pg_cgroups.swap_limit 
-----------------------
 512
(1 row)

-- change swap_limit (should work)
ALTER SYSTEM SET pg_cgroups.swap_limit = 0;
SELECT pg_reload_conf();
 pg_reload_conf 
----------------
 t
(1 row)

SELECT pg_sleep_for('0.3');
 pg_sleep_for 
--------------
 
(1 row)

SHOW pg_cgroups.swap_limit;
 pg_cgroups.swap_limit 
-----------------------
 0
(1 row)

-- lower memory limit (should work)
ALTER SYSTEM SET pg_cgroups.memory_limit = 256;
SELECT pg_reload_conf();
 pg_reload_conf 
----------------
 t
(1 row)

SELECT pg_sleep_for('0.3');
 pg_sleep_for 
--------------
 
(1 row)

SHOW pg_cgroups.memory_limit;
 pg_cgroups.memory_limit 
-------------------------
 256
(1 row)

SHOW pg_cgroups.swap_limit;
 pg_cgroups.swap_limit 
-----------------------
 0
(1 row)

-- raise memory limit (should work)
ALTER SYSTEM SET pg_cgroups.memory_limit = 512;
SELECT pg_reload_conf();
 pg_reload_conf 
----------------
 t
(1 row)

SELECT pg_sleep_for('0.3');
 pg_sleep_for 
--------------
 
(1 row)

SHOW pg_cgroups.memory_limit;
 pg_cgroups.memory_limit 
-------------------------
 512
(1 row)

SHOW pg_cgroups.swap_limit;
 pg_cgroups.swap_limit 
-----------------------
 0
(1 row)

-- set swap limit to -1 (should work)
ALTER SYSTEM SET pg_cgroups.swap_limit = -1;
SELECT pg_reload_conf();
 pg_reload_conf 
----------------
 t
(1 row)

SELECT pg_sleep_for('0.3');
 pg_sleep_for 
--------------
 
(1 row)

SHOW pg_cgroups.swap_limit;
 pg_cgroups.swap_limit 
-----------------------
 -1
(1 row)

-- set swap limit to 0 (should work)
ALTER SYSTEM SET pg_cgroups.swap_limit = 0;
SELECT pg_reload_conf();
 pg_reload_conf 
----------------
 t
(1 row)

SELECT pg_sleep_for('0.3');
 pg_sleep_for 
--------------
 
(1 row)

SHOW pg_cgroups.swap_limit;
 pg_cgroups.swap_limit 
-----------------------
 0
(1 row)

-- set memory limit to 0 (should fail)
ALTER SYSTEM SET pg_cgroups.memory_limit = 0;
ERROR:  invalid value for parameter "pg_cgroups.memory_limit": 0
-- reset all settings
ALTER SYSTEM RESET pg_cgroups.memory_limit;
ALTER SYSTEM RESET pg_cgroups.swap_limit;
SELECT pg_reload_conf();
 pg_reload_conf 
----------------
 t
(1 row)

